libs/aml/src/editor.ts

```
@@ -0,0 +1,316 @@
+
+export type SuggestionKind = 'entity' | 'attribute' | 'pk' | 'index' | 'unique' | 'check' | 'property' | 'value' | 'relation' | 'type' | 'default'
+export type Suggestion = { kind: SuggestionKind, insert: string, label?: string, documentation?: string }
+
+export function computeSuggestions(beforeCursor: string, prevLine: string, db: Database): Suggestion[] {
+    const suggestions: Suggestion[] = []
+    const entities: Entity[] = db.entities || []
+    const attributes: Attribute[] = entities.flatMap(e => flattenAttributes(e.attrs))
+    const relations: Relation[] = db.relations || []
+    const types: Type[] = db.types || []
+    let res: string[] | undefined // storing match result
+    if (res = entityWrittenMatch(beforeCursor)) {
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)